python/cudf/dataframe/column.py

```diff
@@ -603,15 +604,16 @@ def take(self, indices, ignore_index=False):
         if indices.size == 0:
             return self.copy()
 
-        data = cudautils.gather(data=self._data.to_gpu_array(), index=indices)
+        col = cpp_copying.apply_gather_array(self._data.to_gpu_array(),
+                                             indices)
 
         if self._mask:
             mask = self._get_mask_as_column().take(indices).as_mask()
             mask = Buffer(mask)
         else:
             mask = None
```

Comment:
This entire chunk should be replaced with a column gather instead of two array gathers.